tests/conftest.py

```diff
@@ -78,3 +79,16 @@ def set_config(app, name, value):
     app.config[name] = value
     yield
     app.config[name] = old_val
+
+
+@pytest.fixture
+def os_environ():
+    # for use whenever you expect code to edit environment variables
+    old_env = os.environ.copy()
+    os.environ.clear()
```

Comment:
why are we clearing here before the yield? this leads to surprising behaviours (like having to redefine the pytest.ini provided variables like NOTIFY_ENVIRONMENT) and isn't consistent with the api fixture.